GVFS/GVFS.Platform.Windows/WindowsFileSystemVirtualizer.cs

```diff
@@ -1003,31 +1003,9 @@ private HResult QueryFileNameHandler(string virtualPath)
                     if (isDirectory)
                     {
                         GitCommandLineParser gitCommand = new GitCommandLineParser(this.Context.Repository.GVFSLock.GetLockedGitCommand());
-                        if (gitCommand.IsValidGitCommand)
+                        if (gitCommand.IsValidGitCommand || this.FileSystemCallbacks.OnFolderCreated(virtualPath))

```

Comment:
The new logic is subtle - mixing control logic with business functionality (i.e. including `OnFolderCreated(...)` inside the top control structure next to `gitCommand.IsValidGitCommand`). - I think it easier to read if we left the previous structure in place, even if not as compact. Just my thoughts - up to you :)

```
if (gitCommand.IsValidGitCommand)
{
    this.MarkDirectoryAsPlacholder(....)
}
else
{
    if (this.FileSystemCallbacks.OnFolderCreated(virtualPath))
    {
        this.MarkDirectoryAsPlacholder(....)
    }
}
```